### ISSUE TITLE
Question about SHPs calculation

### ISSUE THREAD

**Issue by esrivas17** [DinhHoTongMinh/TomoSAR#10]
Hi, 
I am trying to understand the SHPs calculation implementation.
I have a simple question about `SHP_SelPoint.m`
It seems that you take the SHPs from a pixel only if these are connected to the pixel coordinate within the search window.
 (lines 64-65)
`% connection component`
`LL = bwlabel(SeedPoint);`
`SHP.PixelInd(:,num)=LL(:)==LL(InitRow,InitCol);  `
        
I wonder, why don't take all the pixels within the search window who pass the statistical test as SHPs, regardless if these are not connected spatially to the pixel?

Thanks a lot for your answer,
Erik

**Comment by kjz1997** [comment]
Hello, I think the reason for doing this is because in the real world, adjacent pixels have more similarity in physical representation, and in SAR images, because the resolution is mostly a few meters or more, non adjacent pixels may have sudden changes in physical representation. This is my own understanding. If there is anything inappropriate, please forgive me

**Comment by DinhHoTongMinh** [comment]
You are right in the real-world application. The connectivity constraint assures us that the selected SHPs are statistically similar and form a coherent, local spatial region.
